commit e68d930ac5e7903f2532c08fb2bb4ab0c90f0b1d
Author: Seth Forshee <seth.forshee@canonical.com>
Date:   Tue Feb 9 13:26:34 2016 -0600

    UBUNTU: SAUCE: (namespace) ext4: Add module parameter to enable user namespace mounts
    
    This is still an experimental feature, so disable it by default
    and allow it only when the system administrator supplies the
    userns_mounts=true module parameter.
    
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/fs/ext4/super.c b/fs/ext4/super.c
index b4a547b..6f40c31 100644
--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -110,6 +110,10 @@ static void ext4_clear_request_list(void);
  * transaction start -> page lock(s) -> i_data_sem (rw)
  */
 
+static bool userns_mounts = false;
+module_param(userns_mounts, bool, 0644);
+MODULE_PARM_DESC(userns_mounts, "Allow mounts from unprivileged user namespaces");
+
 #if !defined(CONFIG_EXT2_FS) && !defined(CONFIG_EXT2_FS_MODULE) && defined(CONFIG_EXT4_USE_FOR_EXT2)
 static struct file_system_type ext2_fs_type = {
 	.owner		= THIS_MODULE,
@@ -3294,6 +3298,9 @@ static int ext4_fill_super(struct super_block *sb, void *data, int silent)
 	unsigned int journal_ioprio = DEFAULT_JOURNAL_IOPRIO;
 	ext4_group_t first_not_zeroed;
 
+	if (!userns_mounts && !capable(CAP_SYS_ADMIN))
+		return -EPERM;
+
 	sbi = kzalloc(sizeof(*sbi), GFP_KERNEL);
 	if (!sbi)
 		goto out_free_orig;
